### ISSUE TITLE
Excalidraw Performance Degraded with Large Drawings

### ISSUE THREAD

**Issue by EswaraSatya** [excalidraw/excalidraw#7280]
When creating complex drawings in Excalidraw with a high number of elements (5000+), significant lag and performance issues are experienced. The UI becomes unresponsive and refreshing the drawings takes a long time.

Steps to reproduce:

Open Excalidraw and start a new drawing.
Add a large number of elements to the canvas (shapes, lines, text, etc.) until the element count is over 5000.
Observe the increased lag when panning, zooming and editing the drawing.
Expected behavior:

Excalidraw should perform smoothly and remain responsive even with large, complex drawings. Operations like panning and zooming should not lag noticeably.

Which it is affecting the user experience also

**Comment by Mrazator** [comment]
Yeah, we have additional performance optimizations in mind - in theory, there is still l lot of room for improvement. Firstly we will need to find a way to measure these so that we don't make two steps forward and one backward.

Linking a nice write-up by @jorgedanisc https://github.com/excalidraw/excalidraw/issues/8136

**Comment by tejasPhaveri** [comment]
is there an eta on this? i love this tool for ui mockups but its unusable currently

**Comment by TheKoTech** [comment]
Using excalidraw for wireframes, performance is abysmal even on high-end hardware.
